### ISSUE TITLE
CDN version please

### ISSUE THREAD

**Issue by ttodua** [mattzollinhofer/vue-typeahead-bootstrap#43]
in this github repo I cant find the single js/css to download and include.

Can you put this somewhere on CDN like jsdelivr or cdnjs?
would be quite useful we could include .js file from cdn.
found, 
https://www.jsdelivr.com/package/npm/vue-typeahead-bootstrap

but it says "default file is not set"  - https://i.imgur.com/1tHk5g8.png , and couldnt find the single  .js/.css file to include like we do with all CND-ed js libraries. 

this is main thing that depalizes me from switching from "Alex's" repo to yours, as I can't include the JS (i dont use node.js).

**Comment by ttodua** [comment]
Ah found, good.
https://www.jsdelivr.com/package/npm/vue-typeahead-bootstrap

**Comment by ttodua** [comment]
spent two hours trying to generate the bundle either with nodejs, but simply doesnt work (gives "Cannot use import statement outside a module" error, if making as module, it produces other errors...), no any simple method to use this repo, probably this is only aimed for specific group of programmers to be used and not for general use by most users.

**Comment by mattzollinhofer** [comment]
> it says "default file is not set" 

I don't tend to use CDN-ed js libraries much so I'm not very familiar with how to do this better. Do you have a suggestion?

I'm sorry, I don't fully understand what you're trying to say in your second comment: 
> spent two hours trying to generate the bundle either with nodejs according to your example, but simply doesnt work (gives "Cannot use import statement outside a module" error, if making as module, it produces other errors...), no any simple method to use this repo, probably this is only aimed for specific group of programmers to be used and not for general use by most users.

It sounds like our documentation is wrong or you're having trouble using it. Without more specific examples of your usage it will be difficult to determine. Did you get it to work? If not, please open a second issue with as simple a breakdown as possible. So someone might be able to help. This does not seem to be related to CDN so we won't address it in this issue.

**Comment by ttodua** [comment]
Matt, the problems are related to each other.
for example, even the original repository had a way [see installation](https://github.com/alexurquhart/vue-bootstrap-typeahead#installation) to use the script with `<script src=..url...></script>` approach:
<link href="https://unpkg.com/vue-bootstrap-typeahead/dist/VueBootstrapTypeahead.css" rel="stylesheet">
<script src="https://unpkg.com/vue-bootstrap-typeahead"></script>


but i couldnt find out any way to include this in projects (which are build without nodejs).
Can you tell how to do like the original repo has a way as i describe?

**Comment by ttodua** [comment]
and then, in that case, putting that  `js` and `css` files in jsdelivr package will be what i said "putting them on cdn", so we can direct link them from our projects.

**Comment by mattzollinhofer** [comment]
As I said, I don't use CDN'ed resources much and don't know how to publish to them. I'm very open to it, but it's not at the top of my priority list right now. Feel free to take a look at this repo and issue a PR. 

Maybe @alpharahl has some time to take a look at this?

**Comment by dangerdan** [comment]
I don't need this myself, but this may help: https://www.jsdelivr.com/

**Comment by ttodua** [comment]
Well, I was just not been able to understand how to use. Surely, this seems to be very specifically created only for node.js (and even some node newbies like me are unable to understand how to get the files). Every library (and moreover as i emphasized and is detrimental- the original `vue-typeahead-bootstrap` by alex) has it's direct .js and .css files to be directly used, but this repository doesnt have it, and even i have no clue to find out how to achieve that.

**Comment by RossmacD** [comment]
You would need to build the files to get them via this repo but it looks like jsDelivr does npm too which seems like a better way to do it take a look at this link and you can see it working:
https://cdn.jsdelivr.net/npm/vue-typeahead-bootstrap/


The only thing is that the dist folder does not seem to be here even though its not in the .npmignore?

So i think if you wanted to do this then @mattzollinhofer  would just need to build it and have the dist in npm.
You would end up with something like this then:
https://cdn.jsdelivr.net/npm/vue-typeahead-bootstrap@2.1.2/dist/VueTypeaheadBootsrap.min.js

If you want to generate the js and css yourself for any reason, clone this repository, run `npm install` to get the dependencies, then `npm run build`, it will generate the files and you can get them from the dist folder

**Comment by ttodua** [comment]
@RossmacD  Excellent! thanks for kicking in. yes, I am not Node.Js user so couldnt go over the steps, and just wanted what you said. that's i meant, i couldnt find those files and i've opened this topic, and I think you explained well to author how to do that.

**Comment by wb2you** [comment]
Hi,
I was able to build the js & css file easily after much Googling though.
With Node just run this command in the src/components folder -> "vue build VueTypeaheadBootstrap.vue -t lib".  That then creates the "dist" files for you.

Hope it helps.

Thanks @mattzollinhofer for updating and maintaining this component, it's of great use.
Best regards.

**Comment by ttodua** [comment]
thanks @wb2you  would be nice if repo put the files on jsdelvr too

**Comment by ttodua** [comment]
@wb2you  i couldnt do that. can you write down step by step? 
when I wen to that folder and fired "node vue build VueTypeaheadBootstrap.vue -t lib", it gets errors:


```
C:\Users\xyz\Desktop\vue-typeahead-bootstrap-master\src\components>node vue build VueTypeaheadBootstrap.vue -t lib
internal/modules/cjs/loader.js:960
  throw err;
  ^
Error: Cannot find module 'C:\Users\xyz\Desktop\vue-typeahead-bootstrap-master\src\components\vue'
[90m    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:957:15)[39m
[90m    at Function.Module._load (internal/modules/cjs/loader.js:840:27)[39m
[90m    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12)[39m
[90m    at internal/main/run_main_module.js:18:47[39m {
  code: [32m'MODULE_NOT_FOUND'[39m,
  requireStack: []
}
```

**Comment by wb2you** [comment]
@ttodua I have added a repo on git for the files, hope this helps.
https://github.com/wb2you/Vue-Typeahead-Bootstrap-Dist/

**Comment by wb2you** [comment]
Another hint, when using the "js" version in html, you need to reference to properties with dashes "-" otherwise they remain the default values.
I.e. "minMatchingChars" -> :min-matching-chars
						<vue-typeahead-bootstrap
							:min-matching-chars="0"
							:show-all-results="true"
							:show-on-focus="true"
							size="sm"
							placeholder="type ..."
						  >

**Comment by mattzollinhofer** [comment]
@alpharahl, could you take a look at this? It might be as simple as running `npm run build` as part of the build process, but then does the `dist` directory cause any issues for the vuepress stuff we're doing? Also, do I end up having pretty ugly PRs/commits because the whole dist directory is updated?

**Comment by alpharahl** [comment]
I'm actively looking into this, but noticed @ttodua you need to drop the `node` from `node vue build VueTypeaheadBootstrap.vue -t lib` so you would just run `vue build VueTypeaheadBootstrap.vue -t lib`. I'm still working on making it work myself, however hope to have an update soon.

**Comment by mattzollinhofer** [comment]
@ttodua @wb2you @RossmacD I think we got a version of this working on unpkg, someone want to take a look and see if it's working for them?

https://unpkg.com/vue-typeahead-bootstrap

**Comment by ttodua** [comment]
Thanks, I've tested and it works.
however, there is one however...
I see that js file is more than 100KB (while initial vue-typeahead was 27kb).
is that inevitable ?

**Comment by mattzollinhofer** [comment]
Yeah I noticed that increase in size too. I’m not positive what is causing that. I suspect it’s that we’re using lodash. I might take a quick look, but if it isn’t obvious I’ll probably close this issue and open size optimization as a separate issue.

**Comment by ttodua** [comment]
yes, i have checked lodash too :) but lodash seems (on their site) to be 4kb when zipped...
well, would be nice a smaller size file.. thanks, issue can be closed now.

**Comment by mattzollinhofer** [comment]
I took a quick look at the file size and couldn't find a 5 minute fix, so I created issue #63 to improve the dist file size. If anyone wants to take a crack at it I added some thoughts/details to that issue.

**Comment by mattzollinhofer** [comment]
Last note, this stackoverflow question/answer was helpful to understand the basics: https://stackoverflow.com/questions/31642477/how-to-publish-a-npm-package-with-distribution-files
